### DISCUSSION TITLE
How to make Renovate stop creating PRs with internal beta-releases?

### DISCUSSION THREAD

**Discussion by marverix** [renovatebot/renovate#22396] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35-slim

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Cloud (bitbucket.org)

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I have enabled Renovate for a couple of repositories in our company. The code in the repositories is in Java, and the builder/package manager is Gradle. The artifacts repository is Nexus OSS available only via intranet. On Nexus we have 2 repositories `maven-releases` and `maven-staging`. Our CICD works like that, all feature branches are built and released to `maven-staging`, and on release branches to `maven-releases`. Obviously we have internal dependencies, so we have configured that eg Gradle project A is dependent on Gradle project B, so in Gradle dependencyManagment configuration I provided link to `maven-releases`. No problem there. The last important thing to mention is that we are using everywhere SemVer, and in `maven-releases` we keep production-ready artifacts with versions `MAJOR.MINOR.PATCH`, whereas on `maven-staging` obviously those are NOT production-ready and in version, there is always a suffix `MAJOR.MINOR.PATCH-featurename`. 
The problem is, that Renovate keeps creating PRs to different components with beta packages from `maven-staging`. First I realized that it's because I had everywhere configured both `maven-releases` and `maven-staging` source repositories. But even after I walked thru each repository and removed `maven-staging`, it keeps creating those PRs. I searched in Renovate documentation, but I coudln't find any information how to tell Renovate to either exclude certain registry URL, or to not use versions in format `X.Y.Z-something`, but this 2nd option may be problematic, as I saw yesterday that some public Maven packages are using format `X.Y.Z-final`.

### What you tried so far.

Removing `maven-staging` from dependencyManagment in Gradle project didn't worked.

**Comment by rarkins** [comment, marked as answer]
First of all, the best approach would seem to be to make sure Renovate is not looking up the staging registry if you never want to use it.

The title of this Discussion gives the impression that you think this is Renovate's problem, but it's not. You should be aware that in Gradle versioning `-featurename` does not mean that the release is unstable - only specially designated terms like `-alpha` or `-dev`. I would suggest that you change to use `semver` versioning in Renovate - which then means any suffix indicates unstable - but then you said that you're also mixing in `-final` suffixes. I recommend either:
- Stop using suffixes for stable releases and stick to semver, or
- Stop using non-standard suffixes to indicate unstable

I'm not sure you can easily make either change.

But if you can't work out how to do either of the above, then you could use the `allowedVersions` configuration option with a regex to avoid patterns you don't want. As you seem aware, this could be complex - but the cause of the complexity is the naming convention you've chosen.

**Reply by marverix** [reply]
Thanks for the answer. Let me rephrase: For stable release we never use suffixes. What I meant, is that I saw that some publicly available packages use suffixes, eg. https://mvnrepository.com/artifact/org.hibernate.validator/hibernate-validator . But now I see that they are using `.Final` not `-final`. Sorry, somehow I mixed that out. 
Yeah, so `allowedVersions` would be the solution! Thanks!

But could you explain to me, how Renovate detects the existence of `maven-staging`. As I wrote, I removed `maven-staging` definition from each repo. Is this working like this that Renovate is caching somehow the knowledge about repositories?

**Reply by rarkins** [reply]
It's pretty hard to say from afar, but I don't think it's due to caching, especially unless you have turned on the `repositoryCache` setting. The debug logs from a run may help

**Reply by marverix** [reply]
I have enabled `repositoryCache` :)

**Reply by rarkins** [reply]
Any commit to the base branch should invalidate the config & extract cache. If you clone the repo locally to the latest commit and search for the staging repo string, is it completely missing?
